applications/vanilla/controllers/api/DiscussionsApiController.php

```diff
@@ -645,4 +667,68 @@ public function put_bookmark($id, array $body) {
         $result = $this->discussionByID($id);
         return $out->validate($result);
     }
+
+    /**
+     * Set canonical url for a discussion.
+     *
+     * @param int $id The ID of the discussion.
+     * @param array $body The request body.
+     *       Ex: ["canonicalUrl" => "https://mydomain.com/some+path/"]
+     * @throws NotFoundException If unable to find the discussion.
+     * @return array
+     */
+    public function put_canonical($id, array $body) {
```

Comment:
We're referring to this as a "canonical URL" just about everywhere else, aside from the endpoint. What do you think about renaming this from "/discussions/{id}/canonical" to "/discussions/{id}/canonical-url"?

Reply:
As you wish. Updated